components/candle/challengeService.ts

```diff
@@ -1137,13 +1140,13 @@ export class ChallengeService extends ChallengeRegistry {
             session.challengeContexts[challenge.Id].state = "Start"
         }
 
-        //NOTE: Official will always grant XP to both Location Mastery and the Player Profile
-        const actionXp = challenge.Xp || 0
-        const masteryXp = challenge.Rewards?.MasteryXP || 0
-        const xp = actionXp + masteryXp
-
-        this.grantLocationMasteryXp(masteryXp, actionXp, session, userData)
-        this.grantUserXp(xp, session, userData)
+        controller.progressionService.grantProfileProgression(
+            challenge.Xp,
+            challenge.Rewards?.MasteryXP,
```

Comment:
Please add back the fallbacks to 0, we want to pass on 0 in all cases, never an undefined.